akka-multi-node-testkit/src/main/scala/akka/remote/testconductor/Player.scala

```diff
@@ -88,8 +88,8 @@ trait Player { this: TestConductorExt =>
     import Settings.BarrierTimeout
 
     if (_client ne null) throw new IllegalStateException("TestConductorClient already started")
-    _client = system.actorOf(Props(classOf[ClientFSM], name, controllerAddr), "TestConductorClient")
-    val a = system.actorOf(Player.waiterProps)
+    _client = system.systemActorOf(Props(classOf[ClientFSM], name, controllerAddr), "TestConductorClient")
```

Comment:
@skvithalani Some timing/ordering when shutting down the Player vs terminating the ActorSystem is changed when running this as systemActorOf. It is fixed by changing `stay` to `stop()` here: https://github.com/akka/akka/blob/master/akka-multi-node-testkit/src/main/scala/akka/remote/testconductor/Player.scala#L282

Please add that change.

Use `stop()` for the other TerminateMsg as well, https://github.com/akka/akka/blob/master/akka-multi-node-testkit/src/main/scala/akka/remote/testconductor/Player.scala#L285

Reply:
Sure

Reply:
I have pushed the changes for Player.scala. Wondering if Conductor.scala requires the same changes?

Reply:
That shouldn't be needed (or would be wrong). It's only when it's immediately calls `system.terminate()`

Reply:
sure.